chain/client/src/test_utils/test_loop.rs

```diff
@@ -134,54 +134,57 @@ pub fn route_network_messages_to_client<
             }
             NetworkRequests::Approval { approval_message } => {
                 let other_idx = data.index_for_account(&approval_message.target);
-                if other_idx != idx {
-                    drop(
-                        client_senders[other_idx]
-                            .send_async(BlockApproval(approval_message.approval, PeerId::random())),
-                    );
-                } else {
-                    tracing::warn!("Dropping message to self");
-                }
+                assert_ne!(
```

Comment:
one thing I noticed and still trying to figure out why: when running the test locally if this assertion fails, the test freezes instead of panicing.

Reply:
I remember @robin-near mentioning something about this but I have no memory what the issue was.

Reply:
Robin referred to this issue: https://github.com/near/nearcore/issues/11447

Reply:
Right! That's the one!